### PR TITLE
EZP-27428: Get translated API properties in the language requested

### PR DESCRIPTION
**Pull request by alongosz**
Implements [EZP-27428](https://jira.ez.no/browse/EZP-27428) as continuation of #2002 
---------------
> Target branch: `language_match_api_logic`

This PR is a continuation of #2002

**TODO**:
- [x] Refactor MultiLanguage abstract ValueObjects to interfaces.
- [x] Improve REST Client ContentTypeGroup ValueObject.
- [x] Split `MultiLanguageValueTrait`
- [x] Object state (group) assuming the approach taken on content type (group) is ok
- [x] Add test coverage (currently none to make sure the change does not break anything in existing code)
- [x] Remove TMP commit.

### CODE REVIEW

File: eZ/Publish/SPI/Repository/Values/MultiLanguageDescriptionValueAware.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+namespace eZ\Publish\SPI\Repository\Values;
+
+/**
+ * This is the interface for all ValueObjects implementing translated description logic.
+ *
+ * Provides a uniform way for API consuming logic to generate translated description labels
+ * for API objects.
+ */
+interface MultiLanguageDescriptionValueAware
```
Comment (andrerom) [inline eZ/Publish/SPI/Repository/Values/MultiLanguageDescriptionValueAware.php]: naming.. `Aware` has specific meaning in symfony, so lets perhaps avoid that. Also as it's a interface and not a base value class like Value.php, we can skip imo that to.
So mayme rather just`MultiLanguageDescription`, wdyt @Nattfarinn ?

File: eZ/Publish/Core/REST/Client/Values/ContentType/ContentTypeGroup.php [hunk 2]
```diff
@@ -6,22 +6,28 @@
...
+     * @var string[]
+     */
     protected $descriptions;
 
     public function __construct(array $data = array())
     {
-        foreach ($data as $propertyName => $propertyValue) {
-            $this->$propertyName = $propertyValue;
-        }
+        parent::__construct($data);
```
Comment (andrerom) [inline eZ/Publish/Core/REST/Client/Values/ContentType/ContentTypeGroup.php]: if so you can remove the ctor afaik

File: eZ/Publish/SPI/Repository/Values/MultiLanguageDescriptionValueAware.php (hunk 1 above)
Comment (Nattfarinn) [inline eZ/Publish/SPI/Repository/Values/MultiLanguageDescriptionValueAware.php]: Sure. I made `...Aware` name on the fly, just to point we need to rename it from `...Base`.

Comment (alongosz) [inline eZ/Publish/SPI/Repository/Values/MultiLanguageDescriptionValueAware.php]: Fixed in ac22ecc

File: eZ/Publish/Core/REST/Client/Values/ContentType/ContentTypeGroup.php (hunk 2 above)
Comment (alongosz) [inline eZ/Publish/Core/REST/Client/Values/ContentType/ContentTypeGroup.php]: Hmm, of course. My brain malfunctioned there ;) Fixed in bea8057.

**Comment by alongosz** [comment]
Rebased, last CI: https://travis-ci.org/ezsystems/ezpublish-kernel/builds/239602338
(behat error seems unrelated to this PR)

**Comment by andrerom** [comment]
(But that can be done as separate pr to unblock)

**Comment by alongosz** [comment]
Test on getName is already here: 6b24d7b. 
I moved PR to review because manual tests on content, content type and field definitions went well.
I'm working on tests for getDescription, but if you want to unblock it, I can make another PR.

**Comment by alongosz** [comment]
Update: something's not right, investigating.

**Comment by alongosz** [comment]
Test for ContentType getName and getDescription: b0f0944
There was an error in traits, fixed in 7a0c364

**Comment by andrerom** [comment]
Merging into parent PR to be able to review the result better.
